ZukoAzula2016/src/org/team1540/zukoazula/AutonomousBase.java

```diff
@@ -0,0 +1,159 @@
+package org.team1540.zukoazula;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+
+import ccre.channel.BooleanInput;
+import ccre.channel.FloatInput;
+import ccre.cluck.Cluck;
+import ccre.instinct.AutonomousModeOverException;
+import ccre.instinct.InstinctModeModule;
+import ccre.log.Logger;
+import ccre.rconf.RConf;
+import ccre.rconf.RConf.Entry;
+import ccre.rconf.RConfable;
+import ccre.tuning.TuningContext;
+
+public abstract class AutonomousBase extends InstinctModeModule {
+
+    public AutonomousBase(String modeName) {
+        super(modeName);
+    }
+
+    @Override
+    protected void autonomousMain() throws InterruptedException, AutonomousModeOverException {
+        try {
+            runAutonomous();
+        } finally {
+            Autonomous.allMotors.set(0);
+            Autonomous.fire.set(false);
+            Autonomous.warmup.set(false);
+            Autonomous.intakeArm.set(0);
+        }
+    }
+
+    protected abstract void runAutonomous() throws InterruptedException, AutonomousModeOverException;
+
+    protected void driveForTime(long time, float speed) throws AutonomousModeOverException, InterruptedException {
+        try {
+            Autonomous.allMotors.set(speed);
+            waitForTime(Math.abs(time));
```

Comment:
I don't think we ever used it, but it was there. I'll just remove the absolute value for now.